### PR TITLE
[risk=no][RW-10338] Replace home-page e2e test with equivalent unit tests

### PR DESCRIPTION
**Pull request by Peter-Lavigne**
Replace home-page e2e test with equivalent unit tests. This provides the same level of coverage while running faster and more consistently.

I've left review comments, primarily in the deleted e2e file, justifying each removal.

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [x] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [x] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [x] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: e2e/tests/home/home-page.spec.ts [hunk 1]
```diff
@@ -1,63 +0,0 @@
...
-  const classname = await plusIcon.getProperty<string>('className');
-  expect(classname).toBe('is-solid');
-  const shape = await plusIcon.getAttribute('shape');
-  expect(shape).toBe('plus-circle');
-  const hasShape = await plusIcon.hasAttribute('shape');
-  expect(hasShape).toBe(true);
-  const disabled = await plusIcon.isDisabled();
-  expect(disabled).toBe(false);
-  const cursor = await getStyleValue<string>(page, await plusIcon.asElementHandle(), 'cursor');
-  expect(cursor).toBe('pointer');
```
Comment (Peter-Lavigne) [inline e2e/tests/home/home-page.spec.ts:61]: This is testing that `cursor: 'pointer'` is being passed to the style prop. I think this is self-evident from the style prop and creates a brittle test.

File: e2e/tests/home/home-page.spec.ts [hunk 2]
```diff
@@ -1,63 +0,0 @@
...
-async function checkCreateNewWorkspaceLink(): Promise<void> {
-  const homePage = new HomePage(page);
-  const plusIcon = homePage.getCreateNewWorkspaceLink();
-  expect(plusIcon).toBeTruthy();
-  const classname = await plusIcon.getProperty<string>('className');
-  expect(classname).toBe('is-solid');
-  const shape = await plusIcon.getAttribute('shape');
-  expect(shape).toBe('plus-circle');
-  const hasShape = await plusIcon.hasAttribute('shape');
-  expect(hasShape).toBe(true);
```
Comment (Peter-Lavigne) [inline e2e/tests/home/home-page.spec.ts:57]: These checks all validate that certain attributes have been set on the create button. I think this is self-evident from the code, and that testing it results in brittle tests.

File: e2e/tests/home/home-page.spec.ts [hunk 3]
```diff
@@ -1,63 +0,0 @@
...
-      expect(size).toBeTruthy();
-      expect(size.height).toBeGreaterThan(1);
-      expect(size.width).toBeGreaterThan(1);
-
-      if (width === undefined || height === undefined) {
-        width = size.width; // Initialize width and height with first card element's size, compare with rest cards
-        height = size.height;
-      } else {
-        expect(size.height).toEqual(height);
-        expect(size.width).toEqual(width);
```
Comment (Peter-Lavigne) [inline e2e/tests/home/home-page.spec.ts:31]: Is card size existence and equality something we're concerned will fail? I don't think this is a valuable test unless there's specific logic changing card sizes, which I don't think we have.

File: e2e/tests/home/home-page.spec.ts [hunk 4]
```diff
@@ -1,63 +0,0 @@
...
-      }
-
-      // check workspace name has characters
-      const cardName = await card.getName();
-      await expect(cardName).toMatch(new RegExp(/^[a-zA-Z]+/));
```
Comment (Peter-Lavigne) [inline e2e/tests/home/home-page.spec.ts:36]: Passing the workspace name seems self-evident from the React code.

This seems like the wrong place to validate workspace names. We do allow symbols, so the regex requirement may unexpectedly fail if other tests create workspaces with symbols in names.

File: e2e/tests/home/home-page.spec.ts [hunk 5]
```diff
@@ -1,63 +0,0 @@
...
-
-      // check Workspace Action menu for listed actions
-      const snowmanMenu = await card.getCardSnowmanMenu();
-      // Assumption: test user is workspace Owner.
-      // Check Workspace Actions snowman menu displayes the right set of options.
-      const links = await snowmanMenu.getAllOptionTexts();
-      expect(links).toEqual(expect.arrayContaining(['Share', 'Edit', 'Duplicate', 'Delete']));
```
Comment (Peter-Lavigne) [inline e2e/tests/home/home-page.spec.ts:43]: > // Assumption: test user is workspace Owner.
> // Check Workspace Actions snowman menu displayes the right set of options.

I added equivalent unit tests that make this assumption an explicit requirement and also validate the non-owner case.

File: e2e/tests/home/home-page.spec.ts [hunk 6]
```diff
@@ -1,63 +0,0 @@
...
-
-describe('Home page ui tests', () => {
-  beforeEach(async () => {
-    await signInWithAccessToken(page);
-  });
-
-  test('Check visibility of Workspace cards', async () => {
-    await checkCreateNewWorkspaceLink();
-
-    const allCards = await new WorkspaceCard(page).findAllCards();
```
Comment (Peter-Lavigne) [inline e2e/tests/home/home-page.spec.ts:15]: Interestingly, this test relies on workspaces created by other tests - it doesn't create any itself. If there are no workspaces (or worse, there are workspaces but the cards are failing to appear) the test still passes.

File: ui/src/app/pages/workspace/workspace-card.spec.tsx [hunk 7]
```diff
@@ -34,36 +34,6 @@ describe('WorkspaceCard', () => {
     serverConfigStore.set({ config: { gsuiteDomain: 'abc' } });
   });
 
-  it('disables sharing for non-owners', async () => {
```
Comment (Peter-Lavigne) [inline ui/src/app/pages/workspace/workspace-card.spec.tsx:37]: I moved these checks a level lower into the actions menu tests.

File: ui/src/app/pages/workspace/workspace-card.tsx [hunk 8]
```diff
@@ -244,10 +244,7 @@ export const WorkspaceCard = fp.flow(withNavigation)(
                       />
                     }
                   >
-                    <SnowmanButton
-                      style={{ marginLeft: 0 }}
-                      data-test-id='workspace-card-menu'
```
Comment (Peter-Lavigne) [inline ui/src/app/pages/workspace/workspace-card.tsx:249]: This `data-test-id` is no longer used.

**Review by tarekmamdouh — APPROVED** [review]
lgtm

**Pull request merged**